tests/everflow/everflow_test_utilities.py

```diff
@@ -0,0 +1,292 @@
+    #
+    # We are making sure regular traffic has a dedicated route and does not use
+    # the default route.
+
+    peer_ip, _ = get_neighbor_info(duthost, spine_dest_ports[3])
+
+    add_route(duthost, "30.0.0.1/24", peer_ip)
+
+    yield setup_information
+
+    remove_route(duthost, "30.0.0.1/24", peer_ip)
+
+
+# TODO: This should be refactored to some common area of sonic-mgmt.
+def add_route(duthost, prefix, nexthop):
+    """
+    Add a route to the DUT.
+
+    Args:
+        duthost: DUT fixture
+        prefix: IP prefix for the route
+        nexthop: next hop for the route
+
+    """
+    duthost.shell("vtysh -c \"configure terminal\" -c \"ip route {} {}\"".format(prefix, nexthop))
+
+
+# TODO: This should be refactored to some common area of sonic-mgmt.
+def remove_route(duthost, prefix, nexthop):
+    """
+    Remove a route from the DUT.
+
+    Args:
+        duthost: DUT fixture
+        prefix: IP prefix to remove
+        nexthop: next hop to remove
+
+    """
+    duthost.shell("vtysh -c \"configure terminal\" -c \"no ip route {} {}\"".format(prefix, nexthop))
+
+
+# TODO: This should be refactored to some common area of sonic-mgmt.
+def get_neighbor_info(duthost, dest_port, resolved=True):
+    """
+    Get the IP and MAC of the neighbor on the specified destination port.
+
+    Args:
+        duthost: DUT fixture
+        dest_port: The port for which to gather the neighbor information
+        resolved: Whether to return a resolved route or not
+
+    """
+    if not resolved:
+        return "20.20.20.100", None
+
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)["ansible_facts"]
+
+    for bgp_peer in mg_facts["minigraph_bgp"]:
+        if bgp_peer["name"] == mg_facts["minigraph_neighbors"][dest_port]["name"] and ipaddr.IPAddress(bgp_peer["addr"]).version == 4:
+            peer_ip = bgp_peer["addr"]
+            break
+
+    return peer_ip, duthost.shell("ip neigh show {} | awk -F\" \" \"{{print $5}}\"".format(peer_ip))["stdout"]
+
+
+class BaseEverflowTest(object):
+    """
+    Base class for setting up a set of Everflow tests.
+
+    Contains common methods for setting up the mirror session and describing the
+    mirror and ACL stage for the tests.
+    """
+
+    @pytest.fixture(scope="class")
+    def setup_mirror_session(self, duthost):
+        """
+        Set up a mirror session for Everflow.
+
+        Args:
+            duthost: DUT fixture
+
+        Yields:
+            dict: Information about the mirror session configuration.
+
+        """
+        session_name = "test_session_1"
+        session_src_ip = "1.1.1.1"
+        session_dst_ip = "2.2.2.2"
+        session_ttl = "1"
+        session_dscp = "8"
+
+        if "mellanox" == duthost.facts["asic_type"]:
+            session_gre = 0x8949
+        elif "barefoot" == duthost.facts["asic_type"]:
+            session_gre = 0x22EB
+        else:
+            session_gre = 0x6558
```

Comment:
Probably should define these codes in a more common place so that other tests could share. (Not a change request for this PR).

Reply:
Ack, I'm going to go through and refactor the existing test cases so that all the shared dependencies are in everflow_utilities so I will include this in that effort.